### PR TITLE
adds react-native-sdk

### PR DESCRIPTION
**Pull request by andrewxhill**
Moves library to a [react-native-sdk](https://github.com/textileio/react-native-sdk). It also moves a couple of specific things to modules ([ScreenControl](https://github.com/textileio/react-native-screen-control/), [waitFor](https://github.com/textileio/react-native-wait-for), [protobufs](https://github.com/textileio/react-native-textile-protobufs)).

The setup could be improved if I can figure this out, but it's not a blocker https://github.com/textileio/react-native-sdk/issues/1

TextileImage is still in textile-mobile because I didn't want to deal with figuring out how to create a library for a UI component, but that could be a nice follow-up sooner or later.

### CODE REVIEW

**Comment by andrewxhill** [comment]
might need some ideas on this test failure

it appears as though all the ts types from @textile/react-native-sdk run fine in visual studio sessions for example and compile fine to run the app, but during jest tests the enum types are not working right.  Specifically,

```
export declare enum NotificationType {
    InviteReceivedNotification = "INVITE_RECEIVED",
    AccountPeerJoinedNotification = "ACCOUNT_PEER_JOINED",
    PeerJoinedNotification = "PEER_JOINED",
    PeerLeftNotification = "PEER_LEFT",
    MessageAddedNotification = "MESSAGE_ADDED",
    FilesAddedNotification = "FILES_ADDED",
    CommentAddedNotification = "COMMENT_ADDED",
    LikeAddedNotification = "LIKE_ADDED"
}
```

in textile-mobile now, that gives back a lot of 

`TypeError: Cannot read property 'InviteReceivedNotification' of undefined`

anyplace you use NotificationType.InviteReceivedNotification (or similar) in code touched by the tests. maybe there is a jest/teconfig setting to get that type correctly?

File: App/Sagas/ImageSharingSagas.ts [hunk 1]
```diff
@@ -20,8 +20,8 @@ import TextileNodeActions, { TextileNodeSelectors } from '../Redux/TextileNodeRe
 import { ActionType, getType } from 'typesafe-actions'
 import NavigationService from '../Services/NavigationService'
 import * as CameraRoll from '../Services/CameraRoll'
-import { IMobilePreparedFiles } from '../NativeModules/Textile/pb/textile-go'
-import { waitFor } from './WaitFor'
+import { IMobilePreparedFiles } from '@textile/react-native-protobufs'
+import { waitFor } from '@textile/react-native-wait-for'
```
Comment (asutula) [inline App/Sagas/ImageSharingSagas.ts]: `waitFor` has nothing to do with React Native, it's just a simple JS module. Should probably name it `redux-saga-wait-for` or something.

File: App/Sagas/ImageSharingSagas.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ import TextileNodeActions, { TextileNodeSelectors } from '../Redux/TextileNodeRe
 import { ActionType, getType } from 'typesafe-actions'
 import NavigationService from '../Services/NavigationService'
 import * as CameraRoll from '../Services/CameraRoll'
-import { IMobilePreparedFiles } from '../NativeModules/Textile/pb/textile-go'
-import { waitFor } from './WaitFor'
+import { IMobilePreparedFiles } from '@textile/react-native-protobufs'
```
Comment (asutula) [inline App/Sagas/ImageSharingSagas.ts]: `react-native-sdk` isn't usable without `react-native-protobufs` and `react-native-protobufs` doesn't do anything without `react-native-sdk`, so they could just be combined into a single module.

File: App/Sagas/Migration.ts [hunk 3]
```diff
@@ -14,9 +14,9 @@ import { getSession } from './UploadFile'
 
 import {
   addContact, addThreadFiles, addThread, ThreadInfo, ContactInfo, contact
- } from '../NativeModules/Textile'
-import { IMobilePreparedFiles } from '../NativeModules/Textile/pb/textile-go'
-import { CafeSession } from '../NativeModules/Textile'
+ } from '@textile/react-native-sdk'
+import { IMobilePreparedFiles } from '@textile/react-native-protobufs'
+import { CafeSession } from '@textile/react-native-sdk'
```
Comment (asutula) [inline App/Sagas/Migration.ts]: Multiple import statements from `@textile/react-native-sdk` here, combine them.

File: App/Sagas/NotificationsSagas.ts [hunk 4]
```diff
@@ -14,11 +14,11 @@ import {delay} from 'redux-saga'
 import { call, put, select } from 'redux-saga/effects'
 import { ActionType } from 'typesafe-actions'
 
-import { readAllNotifications as readAll, readNotification, notifications } from '../NativeModules/Textile'
+import { readAllNotifications as readAll, readNotification, notifications } from '@textile/react-native-sdk'
 import {
   NotificationType,
   NotificationInfo
-} from '../NativeModules/Textile'
+} from '@textile/react-native-sdk'
```
Comment (asutula) [inline App/Sagas/NotificationsSagas.ts]: Multiple import statements from @textile/react-native-sdk here, combine them.

File: App/Sagas/ThreadsSagas.ts [hunk 5]
```diff
@@ -14,13 +14,13 @@ import { call, put, select, fork } from 'redux-saga/effects'
...
 import { ActionType } from 'typesafe-actions'
-import { ExternalInvite } from '../NativeModules/Textile'
+import { ExternalInvite } from '@textile/react-native-sdk'
 import {
   addExternalThreadInvite,
   acceptExternalThreadInvite,
   acceptThreadInviteViaNotification,
   addThreadInvite
-} from '../NativeModules/Textile'
+} from '@textile/react-native-sdk'
```
Comment (asutula) [inline App/Sagas/ThreadsSagas.ts]: Multiple import statements from @textile/react-native-sdk here, combine them.

File: App/Services/EventHandlers/TextileNodeEventHandler.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { Store } from 'redux'
 
 import { ILocalPhotoResult } from '../../Models/TextileTypes'
-import {  Update, ThreadUpdate, BlockType, NotificationInfo } from '../../NativeModules/Textile'
-import EventEmitter from '../../NativeModules/Events'
+import {  Update, ThreadUpdate, BlockType, NotificationInfo } from '@textile/react-native-sdk'
+import { Events } from '@textile/react-native-sdk'
```
Comment (asutula) [inline App/Services/EventHandlers/TextileNodeEventHandler.ts]: Multiple import statements from @textile/react-native-sdk here, combine them. Unless there's some reason you want the separate imports?

Comment (andrewxhill) [reply App/Services/EventHandlers/TextileNodeEventHandler.ts]: nah, just did a lot of find/replaces so missed those

File: .circleci/config.yml [hunk 7]
```diff
@@ -1,4 +1,15 @@
...
+
+install-android-tools: &install-android-tools
+    run:
+        name: install android tools
+        command: |
+            sdkmanager --licenses
+            echo y | sdkmanager "build-tools;27.0.3"
+            echo y | sdkmanager "platforms;android-27"
+            echo y | sdkmanager "build-tools;28.0.3"
+            echo y | sdkmanager "platforms;android-28"
```
[the remark this answers is not in the document]
Comment (andrewxhill) [reply .circleci/config.yml]: yeah

[the remark this answers is not in the document]
Comment (sanderpick) [reply .circleci/config.yml]: Some context here... I think the circelci team actually replaced there tagged docker images. The new ones seem to use a newer sdkmanager (I ran into similar building the mobile framework)... the `--licenses` no longer accepts all licenses, so when you go to install one of the platforms, it asks, which is why we have the `echo y` above

**Pull request closed without merging**